src/otx/algo/classification/backbones/timm.py

```diff
@@ -1,57 +1,47 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-"""EfficientNetV2 model.
+"""Timm Backbone Class for OTX classification.
 
 Original papers:
 - 'EfficientNetV2: Smaller Models and Faster Training,' https://arxiv.org/abs/2104.00298,
 - 'Adversarial Examples Improve Image Recognition,' https://arxiv.org/abs/1911.09665.
 """
 from __future__ import annotations
 
-from typing import Literal
-
 import timm
 import torch
 from torch import nn
 
-TimmModelType = Literal[
-    "mobilenetv3_large_100_miil_in21k",
-    "mobilenetv3_large_100_miil",
-    "tresnet_m",
-    "tf_efficientnetv2_s.in21k",
-    "tf_efficientnetv2_s.in21ft1k",
-    "tf_efficientnetv2_m.in21k",
-    "tf_efficientnetv2_m.in21ft1k",
-    "tf_efficientnetv2_b0",
-]
-
 
 class TimmBackbone(nn.Module):
-    """Timm backbone model."""
+    """Timm backbone model.
+
+    Args:
+        model_name (str): The name of the model.
+            You can find available models at timm.list_models() or timm.list_pretrained().
+        pretrained (bool, optional): Whether to load pretrained weights. Defaults to False.
+    """
 
     def __init__(
         self,
-        backbone: TimmModelType,
+        model_name: str,
         pretrained: bool = False,
-        pooling_type: str = "avg",
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.backbone = backbone
+        self.model_name = model_name
         self.pretrained: bool | dict = pretrained
-        self.is_mobilenet = backbone.startswith("mobilenet")
```

Comment:
Is it ok to remove code for mobilenet?

Reply:
This looks like the way it was used in older versions of timm. i think it's unnecessary.